### ISSUE TITLE
Support additonal methods `Other(string)

### ISSUE THREAD

**Issue by sazarkin** [anuragsoni/routes#57]
It will be great to use httpaf `Method.t`
Right now only `Method.standard` supported

**Comment by anuragsoni** [comment]
We can achieve that by switching from an array of routes (current implementation maps the http methods to an array index) to a Map where the keys can then be used as strings. 

Another option to consider (this will be a breaking change) is if remove the http method logic altogether from the library and in-turn use a functor that allows the user to create an instance of the router with the grouping method of their choice. something like:

```ocaml
module Router = Routes.Make (struct
    type t = Httpaf.Method.t
    let compare : t -> t -> int
end)

or 

module Router = Routes.Make (struct
    type t = Cohttp.Method.t
    let compare : t -> t -> int
end)
```

**Comment by anuragsoni** [comment]
Closed by #58

**Comment by anuragsoni** [comment]
This is now available on opam with version 0.5.2
